packages/control/modbusserver.py

```diff
@@ -0,0 +1,147 @@
+from umodbus.utils import log_to_stream
+
+from control import data
+from control.chargemode import Chargemode
+from helpermodules.pub import Pub
+
+
+log = logging.getLogger(__name__)
+
+
+log_to_stream(level=logging.DEBUG)
+data_store = defaultdict(int)
+conf.SIGNED_VALUES = True
+TCPServer.allow_reuse_address = True
+app = get_server(TCPServer, ('0.0.0.0', 1502), RequestHandler)
+
+
+def _form_int32(value, startreg):
+    secondreg = startreg + 1
+    try:
+        binary32 = struct.pack('>l', int(value))
+        high_byte, low_byte = struct.unpack('>hh', binary32)
+        data_store[startreg] = high_byte
+        data_store[secondreg] = low_byte
+    except Exception:
+        log.exception("Fehler beim Füllen der Register")
+        data_store[startreg] = -1
+        data_store[secondreg] = -1
+
+
+def _form_int16(value, startreg):
+    try:
+        value = int(value)
+        if (value > 32767 or value < -32768):
+            raise Exception("Number to big")
+        data_store[startreg] = value
+    except Exception:
+        log.exception("Fehler beim Füllen der Register")
+        data_store[startreg] = -1
+
+
+def _get_pos(number, n):
+    return number // 10**n % 10 - 1
+
+
+@app.route(slave_ids=[1], function_codes=[3, 4], addresses=list(range(0, 32000)))
+def read_data_store(slave_id, function_code, address):
+    """" Return value of address. """
+    if address == 110:
+        ripple_control_receiver = data.data.general_data.data.ripple_control_receiver
+        _form_int16(ripple_control_receiver.r1_active or ripple_control_receiver.r2_active, address)
+    elif address == 111:
+        _form_int16(len(data.data.cp_data), address)
+    elif 300 <= address <= 310:
+        evu_counter = data.data.counter_all_data.get_evu_counter()
+        if address == 300:
+            _form_int32(evu_counter.data.get.power, address)
+        elif 302 <= address <= 304:
+            _form_int16(evu_counter.data.get.currents[address-302]/100, address)
+        elif 305 <= address <= 307:
+            _form_int16(evu_counter.data.get.voltages[address-305]/100, address)
+        elif address == 308:
+            _form_int32(evu_counter.data.get.imported, address)
+        elif address == 310:
+            _form_int32(evu_counter.data.get.exported, address)
+    elif address == 400:
+        _form_int32(data.data.pv_all_data.data.get.power, address)
+    elif address == 402:
+        _form_int32(data.data.pv_all_data.data.get.exported, address)
+    elif address == 500:
+        _form_int32(data.data.bat_all_data.data.get.power, address)
+    elif address == 502:
+        _form_int16(data.data.bat_all_data.data.get.soc, address)
+    elif address == 503:
+        _form_int32(data.data.bat_all_data.data.get.imported, address)
+    elif address == 505:
+        _form_int32(data.data.bat_all_data.data.get.exported, address)
+    elif address > 10099:
+        chargepoint = data.data.cp_data[f"cp{_get_pos(address, 2)}"]
```

Comment:
```suggestion
        chargepoint = data.data.cp_data[f"cp{_get_pos(address, 2) - 1}"]
```
Sollte das Register `10100` nicht Ladepunkt `0` zugeordnet sein?